src/pepr/operator/controllers/config/config.ts

```diff
@@ -0,0 +1,111 @@
+    } catch (e) {
+      log.error(`Failed to decode secret key: ${key}, error: ${e.message}`);
+    }
+  }
+
+  // Handle changes to the Authservice configuration
+  if (
+    decodedConfigData.UDS_CA_CERT !== UDSConfig.caCert ||
+    decodedConfigData.AUTHSERVICE_REDIS_URI !== UDSConfig.authserviceRedisUri
+  ) {
+    UDSConfig.caCert = decodedConfigData.UDS_CA_CERT;
+    UDSConfig.authserviceRedisUri = decodedConfigData.AUTHSERVICE_REDIS_URI;
+
+    // Account for undefined or placeholder values (dev mode)
+    if (
+      !UDSConfig.authserviceRedisUri ||
+      UDSConfig.authserviceRedisUri === "###ZARF_VAR_AUTHSERVICE_REDIS_URI###"
+    ) {
+      UDSConfig.authserviceRedisUri = "";
+    }
+    if (!UDSConfig.caCert || UDSConfig.caCert === "###ZARF_VAR_CA_CERT###") {
+      UDSConfig.caCert = "";
+    }
+    // Validate that the cacert is base64 encoded (it should be)
+    if (UDSConfig.caCert) {
+      try {
+        atob(UDSConfig.caCert);
+      } catch (e) {
+        log.error(
+          "Invalid CA Cert provided in uds-operator-config secret, falling back to no CA Cert",
+        );
+        UDSConfig.caCert = "";
+      }
+    }
+
+    const authserviceUpdate: AuthServiceEvent = {
+      name: "global-config-update",
+      action: Action.UpdateGlobalConfig,
+      // Base64 decode the CA cert before passing to the update function
+      trustedCA: atob(UDSConfig.caCert),
+      redisUri: UDSConfig.authserviceRedisUri,
+    };
+    log.debug("Updating Authservice secret based on change to CA Cert or Redis URI");
+    await reconcileAuthservice(authserviceUpdate);
+  }
+
+  // Handle changes to the kubeApiCidr
+  if (decodedConfigData.KUBEAPI_CIDR !== UDSConfig.kubeApiCidr) {
+    UDSConfig.kubeApiCidr = decodedConfigData.KUBEAPI_CIDR;
+    // This re-runs the "init" function to update netpols if necessary
+    log.debug("Updating KubeAPI network policies based on change to kubeApiCidr");
+    await initAPIServerCIDR();
+  }
+
+  // Handle changes to the kubeNodeCidrs
+  if (decodedConfigData.KUBENODE_CIDRS !== UDSConfig.kubeNodeCidrs) {
+    UDSConfig.kubeNodeCidrs = decodedConfigData.KUBENODE_CIDRS;
+    // This re-runs the "init" function to update netpols if necessary
+    log.debug("Updating KubeNodes network policies based on change to kubeNodeCidrs");
+    await initAllNodesTarget();
+  }
+
+  if (
+    decodedConfigData.UDS_DOMAIN !== UDSConfig.domain ||
+    decodedConfigData.UDS_ADMIN_DOMAIN !== UDSConfig.adminDomain
+  ) {
+    UDSConfig.domain = decodedConfigData.UDS_DOMAIN;
+    UDSConfig.adminDomain = decodedConfigData.UDS_ADMIN_DOMAIN;
+    if (!UDSConfig.domain || UDSConfig.domain === "###ZARF_VAR_DOMAIN###") {
+      UDSConfig.domain = "uds.dev";
+    }
+    if (!UDSConfig.adminDomain || UDSConfig.adminDomain === "###ZARF_VAR_ADMIN_DOMAIN###") {
+      UDSConfig.adminDomain = `admin.${UDSConfig.domain}`;
+    }
+    // todo: Add logic to handle domain changes and update across virtualservices, authservice config, etc
```

Comment:
Create an issue for this todo?

Reply:
https://github.com/defenseunicorns/uds-core/issues/1220